packages/SwingSet/test/bootstrap-relay.js

```diff
@@ -55,10 +63,49 @@ export const buildRootObject = () => {
       // braces to unharden so it can be hardened
       encodePassable(
         Far(label, {
-          ...objectMap(methodReturnValues, v => () => decodePassable(v)),
+          ...objectMap(methodReturnValues, v => {
+            const ret = decodePassable(v);
+            return () => ret;
+          }),
         }),
       ),
 
+    /**
+     * Does the same as makeSimpleRemotable, but also logs the method and
+     * arguments, accessible by calling `getLogForRemotable(remotable)`.
+     *
+     * @param {string} label
+     * @param {Record<string, any>} methodReturnValues
+     */
+    makeLoggingRemotable: (label, methodReturnValues) => {
+      const logs = [];
+      const remotable = Far(label, {
+        ...objectMap(methodReturnValues, (v, k) => {
+          const ret = decodePassable(v);
+          return (...args) => {
+            logs.push([k, ...args]);
+            return ret;
+          };
+        }),
+      });
+      logsByRemotable.set(remotable, logs);
+      return encodePassable(remotable);
+    },
+
+    /**
+     * Returns a copy of `loggingRemotable`s logs.
+     *
+     * @param {Passable} loggingRemotable
+     */
+    getLogForRemotable: loggingRemotable => {
+      const logs = logsByRemotable.get(decodePassable(loggingRemotable));
+      if (!logs) {
+        return undefined;
+      }
```

Comment:
This should fail noisily when the argument is not found in the map.
```suggestion
    getLogForRemotable: remotableSurrogate => {
      const remotable = decodePassable(remotableSurrogate);
      const logs = logsByRemotable.get(remotable) || Fail`unknown remotable: ${q(remotable)}`;
```